File: CoreEmulation/disc/GameManager.cpp
#include "GameManager.h"

namespace Pure3X
{

GameManager::GameManager()
{
}

bool GameManager::AddGame(const std::string& path)
{
    m_games.push_back(path);
    return true;
}

bool GameManager::RemoveGame(const std::string& path)
{
    for (auto it = m_games.begin(); it != m_games.end(); ++it)
    {
        if (*it == path)
        {
            m_games.erase(it);
            return true;
        }
    }

    return false;
}

void GameManager::ClearGames()
{
    m_games.clear();
}

std::size_t GameManager::GetGameCount() const
{
    return m_games.size();
}

const std::vector<std::string>& GameManager::GetGames() const
{
    return m_games;
}

} // namespace Pure3X

File: Cubo3D/app/src/main/java/GameManager.java
package com.pure3x.engine;

import java.util.ArrayList;
import java.util.List;

public class GameManager {

    private final List<String> gameList;

    public GameManager() {
        gameList = new ArrayList<>();
    }

    public void scanGames() {

        gameList.clear();

        gameList.add("=== Pure3XEngenie ===");
        gameList.add("");

        gameList.add("📂 Procurar jogos");
        gameList.add("💾 Memória Interna");
        gameList.add("💾 Cartão SD");
        gameList.add("");

        gameList.add("Formatos suportados:");
        gameList.add("• ISO");
        gameList.add("• PKG");
        gameList.add("• ELF");
        gameList.add("• SELF");
        gameList.add("• PS3_GAME");
        gameList.add("");

        gameList.add("Nenhum jogo encontrado.");
    }

    public List<String> getGames() {
        return gameList;
    }

    public int getGameCount() {
        return gameList.size();
    }

    public void clearGames() {
        gameList.clear();
    }

    public void addGame(String gameName) {
        gameList.add(gameName);
    }
}
